pandas/core/generic.py

```diff
@@ -3004,6 +3004,7 @@ def to_csv(
         quoting: Optional[int] = None,
         quotechar: str = '"',
         line_terminator: Optional[str] = None,
+        lineterminator: Optional[str] = None,
```

Comment:
hmm, this signature is displayed in the documentation so doing it like this could be confusing. (also this function is not kwargs only, so although bonkers, passing arguments as positional arguments would break)

maybe add a **kwargs at the end and document as 'for compatbility with csv module'

then in code something like `kwargs.setdefault('lineterminator', line_terminator)` and pass on kwargs instead of line_terminator to CSVFormatter which just passes them onto csvlib.writer

Reply:
Sounds good! I implemented this idea and added kwargs to the docstrings.

In `to_csv` it's maybe a little awkward
```
kwargs.setdefault("lineterminator", line_terminator)
line_terminator = line_terminator or kwargs["lineterminator"]
```
because we have to feed `line_terminator` to `CSVFormatter` but also we want to keep `line_terminator` explicit for the docs. It works but I'm happy to write this another way if these lines look odd